message_test.go

```diff
@@ -198,3 +198,76 @@ func TestMessageWithMessage(t *testing.T) {
 		t.Errorf("got [%v:%T] want [%v:%T]", got, got, want, want)
 	}
 }
+
+func TestIssue143_Key(t *testing.T) {
+	src := `message Msg {
+  option (option_name) = { [key]: value_name };
+}`
+	p := newParserOn(src)
+	msg := new(Message)
+	p.next()
+	if err := msg.parse(p); err != nil {
+		t.Fatal(err)
+	}
+	name := msg.Elements[0].(*Option).AggregatedConstants[0].Name
+	value := msg.Elements[0].(*Option).AggregatedConstants[0].Literal.Source
+	if got, want := name, "key"; got != want {
+		t.Errorf("got %s want %s", got, want)
+	}
+	if got, want := value, "value_name"; got != want {
+		t.Errorf("got %s want %s", got, want)
+	}
+}
+func TestIssue143_KeyDot(t *testing.T) {
```

Comment:
can we add a case for comments within the option too.

Reply:
that package already has some other tests related to comments in options. Do you have a specific case in mind?